persistence-api/src/main/java/io/stargate/db/query/AsyncQueryExecutor.java

```diff
@@ -0,0 +1,102 @@
+package io.stargate.db.query;
+
+import io.stargate.db.BatchType;
+import io.stargate.db.Parameters;
+import io.stargate.db.Persistence;
+import io.stargate.db.datastore.ResultSet;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.UnaryOperator;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+public interface AsyncQueryExecutor {
```

Comment:
I guess it's a bit of both. The long term goal is that stargate stop using query strings completely, so that `Persistence` also uses query objects. At which point, I want `Persistence.Connection` to also implement this (and more generally that the query builder can build query for either `Persistence` or `DataStore`). Admittedly, there is a bit of work to do before getting there so maybe it's a bit premature, but as this also extract the dependencies a bit better (typically, I'm not using it yet, but that could simplify testing of `AbstractBound#execute` methods)...